engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/IsValueOfTypeNode.java

```diff
@@ -49,6 +49,42 @@ boolean doPolyglotExceptionProxy(Object expectedType, PolyglotExceptionProxy pro
     return isSameObject.execute(expectedType, tpeOfPayload);
   }
 
+  @Specialization(guards = "isAnyType(expectedType)")
+  boolean doAnyType(Object expectedType, Object payload) {
+    return true;
+  }
+
+  @Specialization
+  boolean doLongCheck(Type expectedType, long payload) {
+    var numbers = EnsoContext.get(this).getBuiltins().number();
+    return checkParentTypes(numbers.getSmallInteger(), expectedType);
+  }
+
+  @Specialization
+  boolean doDoubleCheck(Type expectedType, double payload) {
+    var numbers = EnsoContext.get(this).getBuiltins().number();
+    return checkParentTypes(numbers.getDecimal(), expectedType);
+  }
+
+  @Specialization
+  boolean doBigIntegerCheck(Type expectedType, EnsoBigInteger value) {
+    var numbers = EnsoContext.get(this).getBuiltins().number();
+    return checkParentTypes(numbers.getBigInteger(), expectedType);
+  }
+
+  @ExplodeLoop
+  private boolean checkParentTypes(Type actual, Type real) {
```

Comment:
What is the difference between `actual` and `real`. They sound like synonyms to me, looking at the function declaration I do not know what these parameters mean without analysing the implementation and usages.

Maybe we can?
```suggestion
  private boolean checkParentTypes(Type actual, Type expected) {
```

Reply:
Done in 029c297aa